File: src_ROBO720/EX1/KinematicsEX1/src/node2.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Node1 : public rclcpp::Node
{
public:
    Node1() : Node("node2")
    {
        publisher_ = this->create_publisher<std_msgs::msg::String>("topic1", 10);
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            [this]() {
                auto msg = std_msgs::msg::String();
                msg.data = "Hello from node1!";
                RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", msg.data.c_str());
                publisher_->publish(msg);
            });
    }

private:
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<Node1>());
    rclcpp::shutdown();
    return 0;
}

File: src_ROBO720/EX2/fuck_it_we_move_py/fuck_it_we_move_py/node2.py
import rclpy
from rclpy.node import Node

class NodeOne(Node):
    def __init__(self):
        super().__init__('node1')
        self.get_logger().info('Node2 is running!')

def main(args=None):
    rclpy.init(args=args)
    node = NodeOne()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()